sources/packages/web/src/services/UserAuthorizationService.ts

```diff
@@ -0,0 +1,126 @@
+import store from "@/store";
+import {
+  InstitutionRouteUserType,
+  InstitutionUserTypes,
+} from "@/types/contracts/InstitutionRouteMeta";
+import { InstitutionUserAuthRolesAndLocation } from "@/types/contracts/institution/InstitutionUser";
+export class UserAuthorizationService {
+  private static instance: UserAuthorizationService;
+
+  public static get shared(): UserAuthorizationService {
+    return this.instance || (this.instance = new this());
+  }
+
+  public isUserTypeAllowed(
+    allowedTypeList: string[],
+    urlParams?: { locationId?: string },
+    checkAllowedLocation?: { userTypes?: string[] },
+  ) {
+    const isInstitutionAdmin = store.getters["institution/myDetails"]?.isAdmin;
+    const institutionUserTypes: InstitutionUserAuthRolesAndLocation[] =
+      store.getters["institution/myAuthorizationDetails"].authorizations;
+    /* Check is user a Institution Admin and Institution 
+    Admin users has access to the requested page*/
+
+    if (isInstitutionAdmin) {
+      return allowedTypeList.some(
+        type => type === InstitutionRouteUserType.ADMIN,
```

Comment:
If the user is already defined as an admin in the state, why do we need to check the permissions again?

Reply:
no, it's the reverse logic here, if the state of the user is Admin then its checking whether the page is accessible to admin user type.